File: 1217-relative-sort-array/relative-sort-array.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        Map<Integer,Integer> map=new TreeMap<>();
        for(int i:arr1){
            map.put(i,map.getOrDefault(i,0)+1);
        }
        int res[]=new int[arr1.length];
        int k=0;
        for(int i=0;i<arr2.length;i++){
            for(int j=0;j<map.get(arr2[i]);j++){
                res[k++]=arr2[i];
            }
           map.put(arr2[i],0);
        }
        for(int i:map.keySet()){
            int count=map.get(i);
            for(int j=0;j<count;j++)
            {
                res[k++]=i;
            }
        }
        return res;
    }
}

File: 1217-relative-sort-array/relative-sort-array.py
class Solution(object):
    def relativeSortArray(self, arr1, arr2):
        """
        :type arr1: List[int]
        :type arr2: List[int]
        :rtype: List[int]
        """
        mp={}
        for x in arr1:
            mp[x]=mp.get(x,0)+1
        lst=[]
        for x in arr2:
            lst.extend([x]*mp[x])
            mp[x]=0
        
        for x in sorted(mp.keys()):
            if mp[x]!=0:
                lst.extend([x]*mp[x])

        return lst
